bears/vcs/mercurial/HgCommitBear.py

```diff
@@ -0,0 +1,356 @@
+import nltk
+import re
+import shutil
+import os
+
+from urllib.parse import urlparse
+from coalib.bears.GlobalBear import GlobalBear
+from dependency_management.requirements.PipRequirement import PipRequirement
+from coala_utils.ContextManagers import change_directory
+from coalib.misc.Shell import run_shell_command
+from coalib.results.Result import Result
+from coalib.settings.FunctionMetadata import FunctionMetadata
+from coalib.settings.Setting import typed_list
+
+
+class HgCommitBear(GlobalBear):
```

Comment:
you can remove a lot of duplication by doing

```
from bears.vcs.git.GitCommitBear import _CommitBear

class HgCommitBear(_CommitBear):
    ...
```
Try it on a new branch first, testing on your own travis CI.  If causes the bear loader to go nuts, we should file a bug for the bear loader to ignore classes beginning with `_`.

Reply:
Oh
Sure
Thanks a lot 
I will try it on another branch :smile:

Reply:
@jayvdb ,
I tried to extract the common features of both vcs bears into _CommitBear, but when I place the definition of that _CommitBear into the same file as GitCommitBear, the bear loader loads both GitCommitBear and _CommitBear.
So for now I have made separate files for each of them and it seems to run fine.

Also I couldn't find a way to make tests for abstract methods, due to which I couldn't separate the get_metadata() and related functions. 

Any pointers on how to go about this ?
(Also although the latest commit on this PR reflects what I have done to remove redundancy, the Travis CI is passing and CircleCi Failing.... any easier way than ssh-ing into it to debug ? :sweat_smile: )